src/apiMiddleware/index.js

```diff
@@ -0,0 +1,101 @@
+import {
+  DELETE_EXPRESSION,
+  LOGIN_REQUEST,
+  GET_EXPRESSIONS,
+  UPDATE_EXPRESSION_ID,
+  LOGIN_FAILURE,
+  LOGIN_SUCCESS,
+  REGISTER_REQUEST,
+  REGISTER_FAILURE,
+  REGISTER_SUCCESS,
+  FETCH_EXPRESSIONS,
+  REMOVE_EXPRESSION,
+  UPDATE_EXPRESSION,
+  LOGOUT,
+} from 'constants/actionTypes.js';
+import { query } from 'apiMiddleware/query';
+import { LOGIN_URL, EXPRESSIONS_URL, REGISTER_URL } from 'apiMiddleware/config';
+import { getJwt } from 'store/localStorage';
+
+const apiMiddleware = (store) => (next) => (action) => {
+  switch (action.type) {
+    case LOGIN_REQUEST:
```

Comment:
A common practice when dealing with API requests is to use 3 different actions:
1. `${baseActionName}` - the action that this middleware should 'catch'
2. `${baseActionName}_SUCCESS` - the next action in case of success
3. `${baseActionName}_FAILURE` - the next action in case of failure

This should apply for all your API calls. (i.e: LOGIN, REGISTER, FETCH_EXPRESSION, REMOVE_EXPRESSION, UPDATE_EXPRESSION)